external_parser/event_processors/joined_event.h

```diff
@@ -300,6 +300,102 @@ struct ccb_joined_event : public typed_joined_event {
   }
 };
 
+struct DecisionServiceInteractionCats {
+  std::string eventId;
+  std::string timestamp;
+  float action;
+  float pdf_value;
+  float probabilityOfDrop = 0.f;
+  bool skipLearn{false};
+};
+
+struct ca_joined_event : public typed_joined_event {
+  DecisionServiceInteractionCats interaction_data;
+  float reward = 0.0f;
+  float original_reward = 0.0f;
+
+  ~ca_joined_event() = default;
+
+  bool is_skip_learn() const override { return interaction_data.skipLearn; }
+
+  void set_skip_learn(bool sl) override { interaction_data.skipLearn = sl; }
+
+  void set_apprentice_reward() override {
+    if (!std::isnan(interaction_data.action)) {
+      // TODO: default apprenticeReward should come from config
+      // setting to default reward matches current behavior for now
+      reward = original_reward;
+    }
+  }
+
+  void fill_in_label(v_array<example *> &examples) const override {
+
+    if (std::isnan(interaction_data.action)) {
+      VW::io::logger::log_warn("missing action for event [{}]",
+                               interaction_data.eventId);
+      return;
+    }
+
+    if (std::isnan(interaction_data.pdf_value)) {
+      VW::io::logger::log_warn("missing pdf_value for event [{}]",
+                               interaction_data.eventId);
+      return;
+    }
+
+    example *ex = examples[0];
+    ex->l.cb_cont.costs.push_back(
+        {interaction_data.action, 0.f, interaction_data.pdf_value});
+  }
+
+  void set_cost(v_array<example *> &examples, float reward,
+                size_t index = 0) const override {
+    if (std::isnan(interaction_data.action)) {
+      return;
+    }
+
+    examples[0]->l.cb_cont.costs[0].cost = -1.f * reward;
+  }
+
+  bool should_calculate_reward(
+      const std::vector<reward::outcome_event> &outcome_events) {
+    return outcome_events.size() > 0 &&
+           std::any_of(outcome_events.begin(), outcome_events.end(),
+                       [](const reward::outcome_event &o) {
+                         return o.action_taken != true;
+                       });
+    return false;
+  }
+
+  void calc_and_set_cost(
```

Comment:
This will need to be changed once I land my perf change.

calc_and_set_cost was split into calc_cost and set_cost_from_data.